src/metabase/models/dashboard.clj

```diff
@@ -74,19 +74,41 @@
     (assert-valid-parameters dashboard)
     (collection/check-collection-namespace Dashboard (:collection_id dashboard))))
 
-(defn- post-update [dashboard]
-  ;; find any subscriptions for this dashboard and update the name to match
-  (let [affected (db/query
-                  {:select    [:p.id]
+(defn- post-update
+  "Updates the pulses' names and syncs the PulseCards"
+  [dashboard]
+  (let [dashboard-id (:id dashboard)
+        affected (db/query
+                  {:select    [[:p.id :pulse_id] [:pc.card_id :card_id]]
                    :modifiers [:distinct]
                    :from      [[Dashboard :d]]
                    :join      [[DashboardCard :dc] [:= :dc.dashboard_id :d.id]
                                [PulseCard :pc] [:= :pc.dashboard_card_id :dc.id]
                                [Pulse :p] [:= :p.id :pc.pulse_id]]
-                   :where     [:= :d.id (:id dashboard)]})]
-    (when (seq affected)
-      (db/update-where! Pulse {:id [:in (map :id affected)]}
-                        :name (:name dashboard)))))
+                   :where     [:= :d.id dashboard-id]})]
+    (when-let [pulse-ids (seq (distinct (map :pulse_id affected)))]
+      (let [correct-card-ids (->> (db/query {:select [:dc.card_id]
+                                             :modifiers [:distinct]
+                                             :from [[DashboardCard :dc]]
+                                             :where [:= :dc.dashboard_id dashboard-id]})
+                                  (map :card_id)
+                                  (remove nil?)
```

Comment:
Why not just add `[:not= :dc.card_id nil]` to the DB query so we don't have to `(remove nil?)` in Clojure land?